### PR TITLE
Refactoring.

### PR DESCRIPTION
**Pull request by swnsma**
Apply code review changes.

See https://github.com/magento/inventory/pull/3088

### CODE REVIEW

**Comment by sidolov** [comment]
@magento run all tests

**Comment by swnsma** [comment]
@magento run all tests

**Comment by swnsma** [comment]
@magento run all tests

**Comment by swnsma** [comment]
@magento run all tests

**Comment by swnsma** [comment]
@magento run all tests

**Review by novikor — COMMENTED** [review]
Looks fine for now. Let's wait for the test builds.

**Comment by novikor** [comment]
Well, looks promising (not really)

**Comment by swnsma** [comment]
@magento run all tests

File: InventoryBundleProduct/Plugin/InventorySales/IsBundleProductSalablePlugin.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        int $stockId
+    ): bool {
+        try {
+            $types = $this->getProductTypesBySkus->execute([$sku]);
+            if (!isset($types[$sku]) || $types[$sku] !== Type::TYPE_CODE) {
+                return $proceed($sku, $stockId);
+            }
+
+            // @TODO VERY temporary solution untill https://github.com/magento/inventory/pull/3088 will be resolved
+            // Product salability MUST NOT BE CALLED during product load.
```
Comment (novikor) [inline InventoryBundleProduct/Plugin/InventorySales/IsBundleProductSalablePlugin.php]: I think you refer to a wrong PR. #3088 is actually the current PRs base branch.

File: InventoryBundleProduct/Plugin/InventorySales/IsBundleProductSalablePlugin.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        string $sku,
+        int $stockId
+    ): bool {
+        try {
+            $types = $this->getProductTypesBySkus->execute([$sku]);
+            if (!isset($types[$sku]) || $types[$sku] !== Type::TYPE_CODE) {
+                return $proceed($sku, $stockId);
+            }
+
+            // @TODO VERY temporary solution untill https://github.com/magento/inventory/pull/3088 will be resolved
```
Comment (novikor) [inline InventoryBundleProduct/Plugin/InventorySales/IsBundleProductSalablePlugin.php]:
```suggestion
            // @TODO VERY temporary solution until https://github.com/magento/inventory/pull/3088 is resolved
```

File: InventoryCatalog/Plugin/CatalogInventory/Helper/Stock/AdaptAssignStatusToProductPlugin.php [hunk 3]
```diff
@@ -78,6 +79,17 @@ public function beforeAssignStatusToProduct(
         }
 
         try {
+            // @TODO VERY temporary solution untill https://github.com/magento/inventory/pull/3088 will be resolved
```
Comment (novikor) [inline InventoryCatalog/Plugin/CatalogInventory/Helper/Stock/AdaptAssignStatusToProductPlugin.php]: The same as above

File: InventoryBundleProduct/Plugin/InventorySales/IsBundleProductSalablePlugin.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\InventoryBundleProduct\Model\GetBundleProductStockStatus;
+use Magento\InventoryCatalogApi\Model\GetProductTypesBySkusInterface;
+use Magento\InventorySales\Model\IsProductSalableCondition\IsProductSalableConditionChain;
+
+/**
+ * Check if bundle product is salable with bundle options.
+ */
+class IsBundleProductSalablePlugin
```
Comment (novikor) [inline InventoryBundleProduct/Plugin/InventorySales/IsBundleProductSalablePlugin.php]: The namespace already includes `Plugin`.
Yes, we already have a bunch of plugins in the Inventory codebase with a similar naming and it is not described in the technical guidelines, but please take attention anyway.

File: InventoryConfigurableProduct/Plugin/InventorySales/IsConfigurableProductSalablePlugin.php [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryCatalogApi\Model\GetProductTypesBySkusInterface;
+use Magento\InventorySales\Model\IsProductSalableCondition\IsProductSalableConditionChain;
+use Magento\InventorySalesApi\Api\AreProductsSalableInterface;
+
+/**
+ * Check if configurable product is salable.
+ * Moved from chain due to framework limitations (usage of 'AreProductsSalableInterface' provides cyclical dependency).
+ */
+class IsConfigurableProductSalablePlugin
```
Comment (novikor) [inline InventoryConfigurableProduct/Plugin/InventorySales/IsConfigurableProductSalablePlugin.php]: The namespace already contains `Plugin`.

**Review by novikor — CHANGES_REQUESTED** [review]
Looks fine, - but it seems we do not have any active PR and/or issue for disabling `\Magento\CatalogInventory\Observer\AddInventoryDataObserver` (calculating salability status on product load) yet.

**Comment by swnsma** [comment]
@magento run all tests

**Comment by swnsma** [comment]
@magento run all tests

**Comment by novikor** [comment]
@magento run Integration Tests

**Comment by novikor** [comment]
@magento run Integration Tests

**Comment by swnsma** [comment]
@magento run all tests

**Comment by swnsma** [comment]
@magento run Functional Tests CE

**Comment by swnsma** [comment]
@magento run all tests

**Comment by swnsma** [comment]
Failed Integration Tests becomes fixed with removal of _Magento\CatalogInventory\Observer\AddInventoryDataObserver_.
As this observer is going to be removal future, my suggestion is to skip this point for now.

**Review by novikor — APPROVED** [review]

**Pull request merged**